modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java

```diff
@@ -1551,4 +1597,65 @@ else if (msg instanceof GridTaskSessionRequest)
             ? ", task name: " + task.getSession().getTaskName()
             : "";
     }
+
+    /**
+     * Subscription to update the status of tasks.
+     *
+     * <p>NOTE: {@link ComputeGridMonitor#processStatusSnapshots} will be called only on subscription,
+     * then only {@link ComputeGridMonitor#processStatusDiff} will be called.
+     *
+     * @param monitor Task status update monitor.
+     * @throws IllegalStateException If the node is stopped.
+     */
+    public void listenStatusUpdates(ComputeGridMonitor monitor) {
+        lock.writeLock();
+
+        try {
+            if (stopping)
+                throw new IllegalStateException("Failed to add monitor due to grid shutdown: " + monitor);
```

Comment:
Can we throw NodeIsStoppingException instead? We abuse IllegalStateException too much, this is not what it's for